core/eolearn/core/eodata.py

```diff
@@ -796,6 +803,49 @@ def _load_data(file_handler, is_pickle):
             eopatch[feature] = data
         return eopatch
 
+    @staticmethod
+    def walk(filesystem, patch_location):
+        """ Recursively reads a patch_location and returns yelds tuples of (feature_type, feature_name, file_path)
+        """
+        for path in filesystem.listdir(patch_location):
+            if '.' not in path:
+                subdir = fs.path.combine(patch_location, path)
+                for file in filesystem.listdir(subdir):
+                    yield path, file.split('.')[0], fs.path.combine(subdir, file)
+            else:
+                yield path.split('.')[0], None, fs.path.combine(patch_location, path)
+
+    @staticmethod
+    def walk_filtered(filesystem, patch_location, features):
```

Comment:
In case we set features filter, there is still some room to reduce the number of requests that are made to the filesystem. We can talk more in person.

Reply:
Will improve this at later stage, when implementing saving.